pipeline/coordinator.go

```diff
@@ -320,6 +369,21 @@ func (c *Coordinator) runHandlerAsync(job *JobInfo, handler func() (*HandlerOutp
 	})
 }
 
+func (c *Coordinator) runHandlerAsync2(job *JobInfo, handler func() (*HandlerOutput, error)) {
```

Comment:
(needs a better name) This is the similar to `runHandlerAsync()` but only calls `finishJob()` on errors to notify studio of any errors in the copying & probing stage.